CONTRACT.md

```diff
@@ -0,0 +1,96 @@
+---
+authors:
+  - @punitrathore
+issueNumber: 186
+teamSize: 1
+---
+
+# Promises Applied with @zezemanolo #stupendous-goose
+
+## Challenge Rating 7
+
+This goal will likely be within your ZPD if you...
+
+- Are familiar with SQL (the core-sql [goal](https://github.com/GuildCrafts/web-development-js/issues/178) is highly recommended)
+- Are familiar with object oriented programming in JavaScript
+- Are interested in learning about JavaScript Promises
+- Are interested in creating functions which use Promises
+- Are interested in learning how to use the `pg-promise` library
+
+## Description
+
+A theoretical and practical approach to understanding JavaScript Promises.
+
+This goal makes heavy use of external resources: the [Promises Course][promises-course] course on [Udacity](https://www.udacity.com/).
+
+For the first two days you will be working on existing courses and completing tutorials. Then you'll use your skills to complete a set of exercises, and submit your solutions as your artifacts.
+
+Fork the two repos [promise-it-wont-hurt-solutions][promise-it-wont-hurt-solutions] and [pg-promise-exercises][pg-promise-exercises] which contain the exercises you need to complete, and make them your artifacts.
+
+
+## Context
+
+JavaScript is an asynchronous programming language, and if you want want to avoid [callback hell](http://callbackhell.com/) you will have to learn to use Promises.
+
+Promises provide a simpler alternative for executing, composing, and managing asynchronous operations when compared to traditional callback-based approaches. They also allow you to handle asynchronous errors using approaches that are similar to synchronous `try/catch`.
+
+## Specifications
+
+### Day 1 & 2
+- [ ] Complete the [Promises Course][promises-course] on Udacity
+
+### Day 3 & 4
+- [ ] Exercises 1-13 in the [promise-it-wont-hurt-solutions][promise-it-wont-hurt-solutions] are complete and written to the appropriate file:
+  - [ ] `src/01-warmup.js`
```

Comment:
Your comments when submitting to Prrr were really helpful!

In the future, it'd also be helpful to update your contract on what you completed